scripts/unit_tests.sh

```diff
@@ -28,17 +30,30 @@ else
 fi
 
 source /opt/intel/oneapi/setvars.sh intel64 &> /dev/null
-source /etc/profile.d/modules.sh
+source /etc/profile.d/modules.sh &> /dev/null
 export MODULEPATH=$MODULEPATH:/home/pvelesko/modulefiles:/opt/intel/oneapi/modulefiles
 export IGC_EnableDPEmulation=1
 export OverrideDefaultFP64Settings=1
 export CHIP_LOGLEVEL=err
 
-# icpx --version
-# ulimit -a
 sudo /opt/ocl-icd/scripts/igpu_unbind &> /dev/null
 sudo /opt/ocl-icd/scripts/dgpu_unbind &> /dev/null
 
+# Use OpenCL for building/test discovery to prevent Level Zero from being used in multi-thread/multi-process environment
+module load $CLANG
+module load opencl/pocl-cpu-$LLVM
+output=$(clinfo -l 2>&1 | grep "Platform #0")
+if [ $? -ne 0 ]; then
```

Comment:
FYI, the `$?` captures the exit code of grep command.

Reply:
As opposed to the result? I want to fail if "Platform #0" is not found in the output of clinfo

Reply:
Ok, perhaps make the error message more descriptive. E.g. "clinfo failed to execute." -> "Error: no OpenCL platform available."